File: src/Blazor/GitHub/GithubService.cs
using GraphQL;
using GraphQL.Client.Http;
using GraphQL.Client.Serializer.Newtonsoft;
using System.Net.Http.Headers;
using System.Threading.Tasks;

namespace gitskills.Github
{
    public class GithubService
    {
        private GraphQLHttpClient _graphQlClient { get; set; }

        public GithubService(string token)
        {
            _graphQlClient = new GraphQLHttpClient("https://api.github.com/graphql", new NewtonsoftJsonSerializer());

            _graphQlClient.HttpClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", token);
        }

        public async Task<Organization> GetOrgQuery(string OrganizationName)
        {
          orgRequest.Variables = new
          {
              name = OrganizationName
          };
          
          var graphQlResponse = await _graphQlClient.SendQueryAsync<GetOrgQuery>(orgRequest);

          return graphQlResponse.Data.Organization;
        }

        private GraphQLRequest orgRequest = new GraphQLRequest
        {
            Query = @"
query OrganizationQuery ($name: String!) { 
  organization(login: $name){
    name
    membersWithRole(first:100){
      nodes{
        name
        contributionsCollection{
          commitContributionsByRepository{
            repository{
              description
              languages(first:100){
                nodes{
                  name
                }
              }
              repositoryTopics(first:100){
                nodes{
                  topic{
                    name
                  }
                }
              }
            }
            contributions(first:100){
              totalCount
            }
          }
        }
      }
    }
  }
}",
            OperationName = "OrganizationQuery"
        };
    }
}

File: src/Angular/gitskills/src/app/GitHub/github/github.service.ts
import { Injectable, Component, OnInit } from '@angular/core';
import { Apollo } from 'apollo-angular';
import gql from 'graphql-tag';
import { GetOrgQuery_organization, GetOrgQuery_organization_membersWithRole } from './operation-result-types';

@Injectable({
  providedIn: 'root'
})
export class GithubService {

  repos: any;
  loading = true;
  error: any;

  constructor(private apollo: Apollo) { }

  fetchGithub() {
    this.apollo.watchQuery<GetOrgQuery_organization>({
      query: gql `
      query {
        organization(login:"sswconsulting"){
          name
          membersWithRole(first:100){
            nodes{
              name
              contributionsCollection{
                commitContributionsByRepository{
                  repository{
                    description
                    languages(first:100){
                      nodes{
                        name
                      }
                    }
                    repositoryTopics(first:100){
                      nodes{
                        topic{
                          name
                        }
                      }
                    }
                  }
                  contributions(first:100){
                    totalCount
                  }
                }
              }
            }
          }
        }
      }
      `,
    })
    .valueChanges.subscribe(result => {
      this.repos = result.data && result.data.membersWithRole;
      this.loading = result.loading;
      this.error = result.error;
    })
  }
}
